smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -168,7 +171,10 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
         val targetShape =
             model.getShape(shape.target).orElseThrow { CodegenException("Shape not found: ${shape.target}") }
 
-        val targetSymbol = toSymbol(targetShape)
+        var targetSymbol = toSymbol(targetShape)
+        if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT_ZERO_VALUE_V1)) {
+            targetSymbol = targetSymbol.toBuilder().boxed().build()
+        }
```

Comment:
**Nit**: There's no need to make `targetSymbol` mutable:

```kotlin
val targetSymbol = if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT_ZERO_VALUE_V1)) {
    toSymbol(targetShape).toBuilder().boxed().build()
} else {
    toSymbol(targetShape)
}
```

Reply:
Make sense, I will change it accordingly.